src/react-components/ui-root.js

```diff
@@ -228,17 +270,17 @@ class UIRoot extends Component {
       onContinueAfterSignIn();
     };
 
-    this.showDialog(SignInDialog, {
+    this.showNonHistoriedDialog(SignInDialog, {
```

Comment:
Like I said above I would vote all the modals are non historied except that hitting back can close them. Seems simpler than having 2 different types.